src/app/clusters/ota-requestor/DefaultOTARequestor.cpp

```diff
@@ -111,6 +111,30 @@ void DefaultOTARequestor::InitState(intptr_t context)
     DefaultOTARequestor * requestorCore = reinterpret_cast<DefaultOTARequestor *>(context);
     VerifyOrDie(requestorCore != nullptr);
 
+    // Only fetch factory data if vendor ID has not been explicitly set
+    if (!requestorCore->mVendorId.HasValue())
```

Comment:
I think this code could be called in `Init()` just like `mCurrentVersion` initialization. Then the failure could be reported back to the user and the usage of optionals wouldn't be needed anymore.